docs: improve create and modify stacks documentation

tomleah opened the pull request:
<!--
Please read the [Code of Conduct](https://github.com/appsody/website/blob/master/CODE_OF_CONDUCT.md) and the [Contributing Guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md) before opening a pull request.
-->

## Checklist
<!-- For completed items, change [ ] to [x]. -->

- [x] commit message follows [commit guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).
- [x] website builds and runs in production - for information on how to test locally go [here](https://github.com/appsody/website/blob/master/DEVELOPMENT.md).

## Description
<!-- Write a brief description of the changes introduced by this PR -->

## Related Issues
<!--
  Link to the issue that is fixed by this PR (if there is one)
  e.g. Fixes #32, Related to #54, etc.
-->
fixes: https://github.com/appsody/website/issues/79

mhamwala commented on content/docs/stacks/create.md:
```diff
@@ -0,0 +1,100 @@
+
+You **must** include a `Dockerfile-stack` file in the `/image` directory. This file defines the stack and the commands to be used by Appsody. Appsody uses [enviroment variables](/docs/stacks/environment-variables) exposed by the stack image to define these commands.
+
+The `/image/project` directory contains the base of the application. You may decide not to include any functionality for the application but you must include a `Dockerfile` here which will be used by the [`appsody build`](/docs/using-appsody/cli-commands/#appsody-build) command. 
```
we could add the word production here?

> must include a **production** `Dockerfile`

tomleah replied:
Thanks 🙂 Also noticing that I say that "you may decide not to include any functionality for the application" I think I should talk about the value add if you do add functionality/dependencies.

mhamwala replied:
# Yep i agree

mhamwala commented on content/docs/stacks/create.md:
````diff
@@ -0,0 +1,100 @@
+``` 
+
+## Creating a template
+Templates allow for an initial project to be created for the developers. They provide a starter that the developer can expand and adapt as they require.
````
reword suggestion:
> Templates provide an initial application to enable developers to get started with a stack.

tomleah replied:
Great! Thanks

mhamwala commented on content/docs/stacks/create.md:
```diff
@@ -0,0 +1,100 @@
+## Creating a template
+Templates allow for an initial project to be created for the developers. They provide a starter that the developer can expand and adapt as they require.
+
+All templates should be created within `/templates`. Every template is contained within its own directory, `/templates/<stack-name>`.
```
this should be.
`/templates/<template-name>`

tomleah replied:
Good catch, thanks

mhamwala commented on content/docs/stacks/create.md:
````diff
@@ -0,0 +1,100 @@
+```
+A repository is defined by an `index.yaml`. This file lists all stacks and  templates it wants to make avalible.  See the `appsodyhub` [index](https://raw.githubusercontent.com/appsody/stacks/master/index.yaml) as an example.
+
+Each template section must include a link to its `.tar.gz` file. This will be used by `appsody init` to initalize a Appsody project. 
````
`initalize ` --> `initialize`

mhamwala commented on content/docs/stacks/create.md:
````diff
@@ -0,0 +1,100 @@
+my-repo   	file:///Users/foo/index.yaml
+```
+
+Once the repository has been added you can access the stacks by specifying the repository name during a `appsody init`. 
````
**possible reword:**
> Once the repository has been added you can access the stacks in that repo by specifying the repository name when initializing your project:
```
appsody init my-repo/<stack-name>
```

---
***after release of CLI***
Additionally, if you can also set your default repository to point to your own repo  by running the following:
```
appsody init 
```

tomleah replied:
Have used your reword 🙂

Will hold off on the new CLI docs until its cut

mhamwala commented on content/docs/stacks/modify.md:
````diff
@@ -0,0 +1,66 @@
+cd ../templates/<template-name>
+```
+
+3. Edit `.appsody-config.yaml` to use your stack:
````
nit: to use your built stack

mhamwala commented on content/docs/stacks/create.md:
```diff
@@ -25,7 +25,7 @@ The `/image` directory will contain everything that is needed for the stack's im
 You **must** include a `Dockerfile-stack` file in the `/image` directory. This file defines the stack and the commands to be used by Appsody. Appsody uses [enviroment variables](/docs/stacks/environment-variables) exposed by the stack image to define these commands.
 
-The `/image/project` directory contains the base of the application. You may decide not to include any functionality for the application but you must include a `Dockerfile` here which will be used by the [`appsody build`](/docs/using-appsody/cli-commands/#appsody-build) command. 
+The `/image/project` directory contains the base of the application. You may decide not to include any application code here but it is recommended to add some value for the stack, by controlling dependency versions for example. The `project` must include a production `Dockerfile` here which will be used by the [`appsody build`](/docs/using-appsody/cli-commands/#appsody-build) command. 
```
slight reword suggestion: 
> ..but it is recommended to add some value to the stack. For example, by controlling dependency versions..

tomleah replied:
Agreed. This is better, thanks 🙂

neeraj-laad commented on content/docs/stacks/create.md:
```diff
@@ -0,0 +1,100 @@
+## Creating the stack image
+The stack image contains everything that will be common throughout all templates that leverage it. For example, the [`nodejs-express`](https://github.com/appsody/stacks/tree/master/incubator/nodejs-express/image) stack image provides health endpoints and prometheus metrics without developers needing to implement it themselves.
+
+The `/image` directory will contain everything that is needed for the stack's image.
```
How about something like this:

The `/image` directory will contain everything that is needed for the stack's image. You **must** include a `Dockerfile-stack` file in the `/image` directory, which defines how the stack image is built.
 
Stack creators configure [Appsody environment variables](/docs/stacks/environment-variables) in `Dockerfile-stack` to specify the behaviour they expect from the stack throughout the application development lifecycle. `Appsody CLI` and `Appsody controller` inspect these environment variables and then drive the expected behaviour for the developer. 

If a stack image is built upon another stack's image, it will inherit all the Appsody variables from the base stack. It can override the variables it wants to change. This allows users to create stacks with slightly different behaviour while still getting updates from the base stack.

tomleah replied:
Thanks! I've used your suggestion 🙂 I just want to make sure in this doc we are not overwhelming creators with too much information. Maybe we break off to an "advanced" section at some point

neeraj-laad commented on content/docs/stacks/create.md:
```diff
@@ -0,0 +1,100 @@
+
+You **must** include a `Dockerfile-stack` file in the `/image` directory. This file defines the stack and the commands to be used by Appsody. Appsody uses [enviroment variables](/docs/stacks/environment-variables) exposed by the stack image to define these commands.
+
+The `/image/project` directory contains the base of the application. You may decide not to include any application code here but it is recommended to add some value to the stack. For example, by controlling dependency versions. The `project` must include a production `Dockerfile` here which will be used by the [`appsody build`](/docs/using-appsody/cli-commands/#appsody-build) command. 
```
should we use **must at all places?

tomleah replied:
I think so, just to make the compulsory parts stand out

neeraj-laad commented on content/docs/stacks/create.md:
````diff
@@ -0,0 +1,100 @@
+```
+docker build -t <org-name>/<stack-id> -f Dockerfile-stack .
+```
+You can now have access to the stack image to use and test locally.
````
Should we simply point them to our build script and ask to set specific env vars.. like set STACKS_LIST and call build.sh for building and testing? or maybe do both?

tomleah replied:
I think we should do both. I am leaving the build script stuff out for now until we have documentation on how to build and test using the scripts. I'm not sure if this belongs in the stacks repo or the website, or both

neeraj-laad commented on content/docs/stacks/create.md:
````diff
@@ -0,0 +1,100 @@
+```
+$ appsody repo list
+NAME      	URL                                                               
+appsodyhub	https://raw.githubusercontent.com/appsody/stacks/master/index.yaml
````
perhaps update the url with new one or just use some text like <url for repo index>

neeraj-laad commented on content/docs/stacks/create.md:
````diff
@@ -0,0 +1,100 @@
+
+Once the repository has been added you can access the stacks in that repo by specifying the repository name when initializing your project:
+```
+appsody init my-repo/<stack-name>
````
We could also suggest building a local index using our build scripts?

neeraj-laad commented on content/docs/stacks/modify.md:
````diff
@@ -0,0 +1,66 @@
+```
+
+You should now make sure that all the stack templates work with the modified stack image.
+
````
same as before we should ask them to use the build process. so when we add linters etc they utilize those for dev too

neeraj-laad commented on content/docs/stacks/modify.md:
````diff
@@ -0,0 +1,66 @@
+appsody run 
+``` 
+
+**Note:** You should test other Appsody commands that the stack supports also.
````
test all

neeraj-laad commented on content/docs/stacks/modify.md:
```diff
@@ -0,0 +1,66 @@
+   * Any restrictions/limitations of the stack
+
+## Modifying a template
+Modifying a template is simple and allows the developer to get started with their applications, quicker. Whilst modifying templates consider if the functionality may be better placed in the stack image itself to affect all templates.
```
I like how u have worded this part..

tomleah commented:
@neeraj-laad I've addressed some of your comments but left parts that reference the stacks ci system. I think we need to discuss where these docs should go and where the user will be directed.
